### DISCUSSION TITLE
OnFrameLoadEnd doesn't persist hooked between page loads

### DISCUSSION THREAD

**Discussion by lofti198** [cefsharp/CefSharp#4175] [General]
Using ChromiumWebBrowser for WinForms. When I load page first time I get into OnFrameLoadEnd handler.

public Form1()
        {
           InitializeComponent();
           
            WindowState = FormWindowState.Maximized;

            browser = new ChromiumWebBrowser("www.google.com");
            toolStripContainer1.ContentPanel.Controls.Add(browser);
            
            
            browser.FrameLoadEnd += OnFrameLoadEnd;
            browser.MenuHandler = new CustomMenuHandler();

            browser.RequestHandler = new CustomRequestHandler();
    }`
`

However when I load the next page using `browser.Load("http://site.net")`, it doest not fire. The same story with JavascriptMessageReceived . What am I doing wrong?

**Comment by amaitland** [comment]
For testing purposes comment out your CustomRequestHandler, just in case it's blocking pages from loading. 

Also check the log file for errors see https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#log-file

If that doesn't work then you'll need to provide more detail to your code example. OnFrameLoadEnd event handler for starters. Your code doesn't subscribe to the JavascriptMessageReceived event from what I can tell.

Overall there's too many pieces missing from your example code for me to say what's going on.

**Comment by lofti198** [comment]
Hi! I figured out , that some websites have infinite loading time. Trying for instance for https://kolchaka.net - I`m not getting loadingstatechanged the second time (for e.isLoading==faslse) and FrameLoadEnd. I guess I can do nothing about this, however maybe there is any workaround, do I could hook JavascriptMessageReceived to the webpage at least after partial load (with JavascriptMessageReceived I will then catch user mouseclicks on webpage).

`
public Form1()
        {
            
            InitializeComponent();

            WindowState = FormWindowState.Maximized;


            ChromiumWebBrowser browser = new ChromiumWebBrowser("https://kolchaka.net");
            
            browser.LoadingStateChanged += Browser_LoadingStateChanged;
            browser.FrameLoadEnd += Browser_FrameLoadEnd;
            toolStripContainer1.ContentPanel.Controls.Add(browser);
        }

        private void Browser_LoadingStateChanged(object sender, CefSharp.LoadingStateChangedEventArgs e)
        {
            Debug.WriteLine($"{DateTime.Now}: Frame loading state: {e.IsLoading}");
            Invoke((MethodInvoker) delegate { button1.Enabled = !e.IsLoading; });
        }
        private void Browser_FrameLoadEnd(object sender, CefSharp.FrameLoadEndEventArgs e)
        {
            Debug.WriteLine($"{DateTime.Now}: Frame loaded {e.Url}");  
        }

`

**Reply by amaitland** [reply]
Does your website ever finish loading in Chrome? I'd suggest opening DevTools to look at the Network tab to get a better understanding of what's going on.

**Comment by lofti198** [comment]
also found this in debug console:

ERROR:browser_info_manager.cc(559)] Timeout of new browser info response for frame
